session.go

```diff
@@ -51,6 +51,15 @@ type Session interface {
 	// Warnings returns a list of warnings generated for the session on connect/auth
 	Warnings() []error
 
+	// ListenAndForward creates a new Tunnel which will listen for new inbound
+	// connections. Connections on this tunnel are automatically forwarded to
+	// the provided URL.
+	ListenAndForward(ctx context.Context, backend *url.URL, cfg config.Tunnel) (Forwarder, error)
+
+	// ListenAndServeHTTP creates a new Tunnel to serve as a backend for an HTTP server. Connections will be
+	// forwarded to the provided HTTP server.
+	ListenAndServeHTTP(ctx context.Context, cfg config.Tunnel, server *http.Server) (Tunnel, error)
```

Comment:
Should we maybe also have a `ListenAndHandleHTTP` that takes an `http.Handler`, and wraps it in a default http.Server for you? Kinda like with the now-deprecated options.

Reply:
Yeah, I had that thought, wasn't sure if we wanted to supported `http.Handler` or not. I'll add it in 👍🏼